tools/toolchain/scripts/stage7/install_hdf5.sh

```diff
@@ -6,8 +6,8 @@
 [ "${BASH_SOURCE[0]}" ] && SCRIPT_NAME="${BASH_SOURCE[0]}" || SCRIPT_NAME=$0
 SCRIPT_DIR="$(cd "$(dirname "$SCRIPT_NAME")/.." && pwd -P)"
 
-hdf5_ver="1.14.2"
-hdf5_sha256="ea3c5e257ef322af5e77fc1e52ead3ad6bf3bb4ac06480dd17ee3900d7a24cfb"
+hdf5_ver="1.14.3"
+hdf5_sha256="9425f224ed75d1280bb46d6f26923dd938f9040e7eaebf57e66ec7357c08f917"
```

Comment:
As a matter of fact, I realized yesterday night that I changed that. On the master branch we are still at 1.14.2, but on the cp2k downloads repo I found that [1.14.3 is uploaded](https://www.cp2k.org/static/downloads/hdf5-1.14.3.tar.bz2). Yesterday I tried again to revert to 1.14.2 and I remembered why I changed: with my machine (arch linux, gcc 14.2.1, and the rest from the toolchain) I cannot manage to compile 1.14.2.
On the other hand, I can compile 1.14.3 and also everything works with my system-wide hdf5, which is 1.14.5.
As far as I am concerned, we can move on to 1.14.5.

Notice also that I changed the hdf5 script to search for system-wide library, as in arch linux (maybe in other distros as well), using simply https://github.com/cp2k/cp2k/blob/2de90c3493f93ab5519e4c813f6f1ead4f3a97fc/tools/toolchain/scripts/stage7/install_hdf5.sh#L58-L61
does not work, as the libraries `h5cc` returns `/usr/lib`, ending up with paths `/usr/lib/lib` and `/usr/lib/include` which are wrong. The `lib` and `include` paths on arch are `/usr/lib` and `/usr/include`, hence my changes.

Reply:
Ok, thanks. So if I update the hdf5 version used in this PR, then it will be also changed in master once merged and I don't know whether that is gonna break other libraries which depend on hdf5, like SIRIUS. Is this anyhow fine?

Reply:
Since it's only a patch level upgrade, I would not expect any issues with other libraries. 
However, if you'd like to play it safe then you could open a separate PR for just the HDF5 upgrade and afterwards return to this PR.

Reply:
I didn't think about that. It's true that for a patch level upgrade it should not be problematic.